specs/NonClientHitTestKind.md

```diff
@@ -0,0 +1,224 @@
+Hit Test Kind
+===
+
+# Background
+The ability to ask WebView2 for information on a hit test is an important feature for 
+visually hosted apps. It can be used to support features related to mouse activity in 
+regions that are traditionally Non-Client areas. Features like Draggable Regions, 
+Resize, Minimize/Maximize functionality, and more.
+
+
+# Examples
+## Draggable regions (HWND)
+Draggable regions are marked regions on a webpage that will move the app window when the 
+user clicks and drags on them. It also opens up the system menu if right-clicked. This 
+API is designed to provide draggable regions support for apps that are directly hosting 
+the CoreWebView2 via visual hosting and for UI framework WebView2 controls that use visual 
+hosting, like the WinUI2 and WinUI3 WebView2 control.If your app uses windowed hosting, 
+draggable regions are supported by default, no need to follow this guide.
+
+NOTE: Your app may already be forwarding mouse messages to WebView2, if so the bulk 
+of the code below is likely already in your app. One thing you need to do specifically 
+for this feature is to add the check for w_param == HTCAPTION as a clause for capturing 
+the mouse and forwarding messages. It is necessary for the system menu to show. If you 
+choose not to include this you would have to handle the logic for showing the system menu 
+yourself.
+
+NOTE: This API is not currently supported for environments that use a CoreWindow like UWP
+
+#### Win32 C++
+```cpp
+LRESULT CALLBACK WndProc(HWND hwnd, UINT message, WPARAM wParam, LPARAM lParam)
+{
+....
+// The following code forwards mouse messages to the WebView2 - Start
+
+// used to control when messages are forwarded to WebView2
+static bool isCapturingMouse = false; 
+
+// handle the range of mouse messages & WM_NCRBUTTONDOWN
+if (message >= WM_MOUSEFIRST && message <= WM_MOUSELAST || message == WM_NCRBUTTONDOWN) 
```

Comment:
Will that be part of this API spec?

Reply:
It should be, since this API spec is adding the need to pass this value to CWV2.

Unless we can have the app pass through just RBUTTON_DOWN and have WV2 know this occurred over an area that is HTCAPTION (setting the internal state), so the next RBUTTON_UP can result in showing the system menu.

Reply:
Added a new enum value for NC_RBUTTONDOWN.